src/foundry/client/pixi/webgl/shaders/lighting/base-lighting.d.mts

```diff
@@ -0,0 +1,182 @@
+export {};
+declare global {
+  interface ShaderTechnique {
+    /** The numeric identifier of the technique */
+    id: number;
+
+    /** The localization string that labels the technique */
+    label: string;
+
+    /**  The coloration shader fragment when the technique is used */
+    coloration: string | undefined;
+
+    /** The illumination shader fragment when the technique is used */
+    illumination: string | undefined;
+
+    /** The background shader fragment when the technique is used */
+    background: string | undefined;
+  }
+
+  /**
+   * This class defines an interface which all adaptive lighting shaders extend.
+   */
+  class AdaptiveLightingShader extends AbstractBaseShader {
+    /**
+     * Has this lighting shader a forced default color?
+     * @defaultValue `false`
+     */
+    static forceDefaultColor: boolean;
+
+    /**
+     * Called before rendering.
+     * */
+    update(): void;
+
+    /**
+     * Common attributes for vertex shaders.
+     */
+    static VERTEX_ATTRIBUTES: string;
+
+    /**
+     * Common uniforms for vertex shaders.
+     */
+    static VERTEX_UNIFORMS: string;
+
+    /**
+     * Common varyings shared by vertex and fragment shaders.
+     */
+    static VERTEX_FRAGMENT_VARYINGS: string;
+
+    /**
+     * Common functions used by the vertex shaders.
+     * @remarks Foundry labeled as abstract
+     */
+    static VERTEX_FUNCTIONS: string;
+
+    /**
+     * Common uniforms shared by fragment shaders.
+     */
+    static FRAGMENT_UNIFORMS: string;
+
+    /**
+     * Common functions used by the fragment shaders.
+     * @remarks Foundry labeled as abstract but provided a default value?
+     */
+    static FRAGMENT_FUNCTIONS: string;
+
+    static override CONSTANTS: string;
+
+    static override vertexShader: string;
+
+    /**
+     * Construct adaptive shader according to shader type
+     * @param shaderType - shader type to construct : coloration, illumination, background, etc.
+     * @returns the constructed shader adaptive block
+     */
+    static getShaderTechniques(shaderType: AdaptiveLightingShader.ShaderTypes): string;
+
+    /**
+     * The coloration technique coloration shader fragment
+     */
+    static get COLORATION_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The coloration technique illumination shader fragment
+     */
+    static get ILLUMINATION_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The coloration technique background shader fragment
+     */
+    static get BACKGROUND_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
```

Comment:
If you want to be more DRY my recommendation is to add a type somewhere for reference. Stuff like `ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;`  is just too verbose.

Reply:
I don't know the rules around when to let a type be global vs namespaced, so I made this one namespaced, which leads to lines *almost* as long unfortunately. Done.